spec/controllers/orders_controller_spec.rb

```diff
@@ -0,0 +1,216 @@
+require "rails_helper"
+include SessionsHelper
+RSpec.describe OrdersController, type: :controller do
+  let(:category){
+    Category.create!(
+      name: "Category 1",
+      description: "Description 1",
+      duration: 2,
+    )
+  }
+  let(:tour){
+    Tour.create!(
+      name: "Tour number 1",
+      description: "Description tour 1",
+      start_date: Date.tomorrow,
+      end_date: Date.tomorrow,
+      cost: 200000,
+      visit_location: "Vinh Phuc",
+      start_location: "Ha Noi",
+      content: "<div>Lich Trinh</div>",
+      tour_guide_cost: 100000,
+      category: category,
+      image: {
+        io: URI.open("https://hinhanhonline.com/Images/Album/DulichVietNam/vinh-ha-long-01.jpg"),
+        filename: "image_1.jpg",
+        content_type: "image/jpeg"
+      }
+    )
+  }
+  let(:user){
+    User.create!(
+      phone: "0123456789",
+      email: "lecuong@gmail.com",
+      address: "120 Tran Binh",
+      password: "123456",
+      password_confirmation: "123456",
+      name: "cuong",
+      role: "user",
+    )
+  }
+  let(:order){
+    Order.create!(
+      tour: tour,
+      user: user,
+      contact_name: "Nguyen Duc Huy",
+      contact_phone: "0975222888",
+      contact_address: "133 Xuan Thuy",
+      amount_member: "2",
+      note: "Ghi chu",
+      tour_guide: "1",
+      total_cost: "8150000"
+    )
+  }
+
+  describe "GET #index" do
+    context "when log in" do
+      before do
+        log_in user
+      end
+
+      it "render index template" do
+        get :index, params: { user_id: user.id }
+        expect(response).to render_template(:index)
+      end
+
+      it "assign @orders" do
+        get :index, params: { user_id: user.id }
+        expect(assigns(:orders)).to eq([order])
+      end
+    end
+    context "when not log in" do
+      before do
+        log_out if logged_in?
+      end
+
+      it "flash not log in" do
+        get :index, params: { user_id: user.id }
+        expect(flash[:danger]).to eq I18n.t("orders.flash.not_logged_in")
+      end
+    end
+  end
+
+  describe "GET #new" do
+    context "when Start date is before Today" do
+      before do
```

Comment:
phân biệt before và before_all

Reply:
before (before each) sẽ được chạy trước khi chạy các example
before all sẽ chạy 1 lần trước khi chạy toàn bộ các example, before all chạy trước before